templates/dockerize.yml

```diff
@@ -0,0 +1,119 @@
+# This template is expects exactly ONE Dockerfile in the microservice repo
+# This template utilizes an external dependency called mvn-version-1.2.1 to 
+# extract the version of the Maven pom.xml file, which is then used as the
+# tag for the docker image that is pushed.
+
+parameters:
+
+  # project configuration parameters:
+- name: imageName                   # <-- e.g., project-registry-gateway
+  type: string
+- name: containerRepo               # <-- e.g., myproject.azurecr.io, myDockerHub
+  type: string
+- name: containerServiceConnection  # <-- name of the service connection associated
+  type: string                      #     with the above repo
+- name: containerUrl                # <-- URL to the associated repo
+  type: string
+- name: version                     # <-- you should only use this parameter if the value of
+  type: string                      #     useDynamicVersion is set to false, otherwise it is 
+  default: undefined                #     overwritten dynamically using mvn-version-1.2.1
+- name: useDynamicVersion           # <-- if this is set to true then it will use the value
+  type: boolean                     #     specified by parameters.version rather than the version
+  default: true                     #     pulled from the pom.xml file (default behavior)
+
+  # external access parameters:
+- name: discordChannel
+  type: string
+- name: discordKey
+  type: string
+
+steps:
+- ${{ if eq(parameters.useDynamicVersion, true) }}:
+  - script: |
+      MVN_VERSION_DOWNLOAD='https://github.com/emcrisostomo/mvn-version/archive/refs/tags/1.2.1.tar.gz'
+      mkdir mvn-version && 
+      cd mvn-version &&
+      mkdir 1.2.1 &&
+      curl -L $MVN_VERSION_DOWNLOAD |
+        tee 1.2.1.tar.gz |
+        tar zx -C 1.2.1
+    workingDirectory: $(Build.SourcesDirectory)
+    displayName: download mvn-version
+
+  - script: |
+      sudo apt-get update &&
+      sudo apt-get install zsh -y &&
+      sudo apt-get install xmlstarlet -y &&
+      pwd && ls &&
+      sudo bash autogen.sh &&
+      sudo ./configure &&
+      sudo make &&
+      sudo make install
```

Comment:
didn't we settle on downloading the binary rather than building the source?

Reply:
It may be that I am missing something obvious, but I couldn't find a 'binary', but there was a tar, so I used that.

Reply:
"binary" is what we call executable files as opposed to source code. that's the origin of the "bin/" folder you sometimes see. in this case the binary is in the tar (also called a "tarball" for some reason).

Reply:
Did you actually see an executable in the tar and I am just blind or naive?  
All I can find are source code files.  It appears that the contents of the tar file are identical to the contents of the repo.

Reply:
well, i said that, but on closer inspection this package is only released as source code and you're just mean to build it yourself like you're doing. that's a little inconvenient to do every build but it works. i hope it's not too slow.

Reply:
It's pretty fast.